docs/source/models.rst

```diff
@@ -81,6 +81,98 @@ time_created
+endpoint with mlflow model serve.
+
+Below is an example  of model signature for classification model build over well known iris dataset.
+The input is 4 named numeric columns, output is unnamed integer specifying the predicted class:
+
+.. code-block:: yaml
+
+  signature:
+      inputs: '[{"name": "sepal length (cm)", "type": "double"}, {"name": "sepal width
+        (cm)", "type": "double"}, {"name": "petal length (cm)", "type": "double"}, {"name":
+        "petal width (cm)", "type": "double"}]'
+      outputs: [{"type": "integer"}]
+
+Signature Enforcement
+~~~~~~~~~~~~~~~~~~~~~
+MLflow will enforce input schema when scoring models that include signature. The schema enforcement
+checks input column ordering and column types and will raise an exception if the input is not
+compatible. This enforcement is applied in MLflow before calling the underlying model
+implementation. Note that this enforcement only applies when using MLflow model deployment tools or
+when loading models as ``python_function``. In particular, it is not applied to models that are
+loaded in their native format (e.g. by calling mlflow.sklearn.load_model).
+
+Column Ordering Enforcement
+"""""""""""""""""""""""""""
+The input columns are checked against the model signature. If there are any missing columns,
+MLflow will raise an exception. Extra columns that were not declared in the signature will be
+ignored. If the input schema in the signature defines column names, column matching is done by name
+and the columns are reordered to match the signature. If the input schema does not have column
+names, matching is done by position (i.e. MLflow will only check the number of columns).
+
+Column Type Enforcement
+"""""""""""""""""""""""
+The input column types are checked against the signature. MLflow will perform safe type conversions
+if necessary. Generally, only upcasts (e.g. integer -> long or float -> double) are considered to be
+safe. If the types can not be made compatible, MLflow will raise an error.
+
+How To Log Models With Signature
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The model signature can be :py:func:`inferred <mlflow.models.infer_signature>`
+from datasets with valid model input (e.g. the training dataset with target
+column omitted) and valid model output (e.g. model predictions generated on
+the training dataset). To include signature with your model, add it to the appropriate log_model
+call, e.g. :py:func:`sklearn.log_model <mlflow.sklearn.log_model>`:
+
+.. code-block:: python
+
+    from mlflow.models.signature import infer_signature
+    train = df.drop_column("target_label")
+    predictions = ... # compute model predictions
+    signature = infer_signature(train, predictions)
+    mlflow.sklearn.log_model(..., signature=signature)
+
+
+Model Input Example
+^^^^^^^^^^^^^^^^^^^
+Input example provides one or several instances of valid model input. The example can be used as a
```

Comment:
By "several examples", do we mean that the example might be a batch of records? In that case, it may be clearer to say something like:

```suggestion
A model input example provides an instance of a valid model input. This may be a single record or a batch of records.
```